### PR TITLE
clang-format: make public _with_style so that OnceCell isn't required

### PR DESCRIPTION
**Pull request by ahayzen-kdab**
Related to #3

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -62,9 +62,25 @@ pub enum ClangFormatError {
...
+/// "#;
+/// let output = clang_format_with_style(input, ClangFormatStyle::Mozilla);
+/// assert!(output.is_ok());
+/// assert_eq!(output.unwrap(), "\nstruct Test\n{};\n");
+/// # }
+/// ```
+pub fn clang_format_with_style(
     input: &str,
-    style: &ClangFormatStyle,
+    style: ClangFormatStyle,
```
Comment (LeonMatthesKDAB) [inline src/lib.rs:83]: why did this change to consuming the style?

Comment (ahayzen-kdab) [reply src/lib.rs:83]: When making the `clang_format_with_style` public i wondered which of the following felt more natural ?

Maybe be-ref is fine though, i can change it back :-) 

```rust
clang_format_with_style(input, ClangFormatStyle::Mozilla);
or
clang_format_with_style(input, &ClangFormatStyle::Mozilla);
```

Comment (ahayzen-kdab) [reply src/lib.rs]: Reverted back to by-ref

**Review by LeonMatthesKDAB — APPROVED** [review]
LGTM

**Pull request merged**